README.md

```diff
@@ -59,6 +60,31 @@ Other supported formats are listed below.
 * `Workstation ID` - The workstation name (default is the host name)
 * `ApplicationIntent` - Can be given the value `ReadOnly` to initiate a read-only connection to an Availability Group listener. The `database` must be specified when connecting with `Application Intent` set to `ReadOnly`.
 
+### Kerberos Active Directory authentication outside Windows
+The package supports authentication via 3 methods.
+
+* Keytabs - Specify the username, keytab file, the krb5.conf file, and realm.
+
+      authenticator=krb5;server=DatabaseServerName;database=DBName;user id=MyUserName;realm=domain.com;krb5conffile=/etc/krb5.conf;keytabfile=~/MyUserName.keytab
+
+* Credential Cache - Specify the krb5.conf file path and credential cache file path.
+
+      authenticator=krb5;server=DatabaseServerName;database=DBName;krb5conffile=/etc/krb5.conf;krbcache=~/MyUserNameCachedCreds
+
+* Raw credentials - Specity krb5.confg, Username, Password and Realm.
+
+      authenticator=krb5;server=DatabaseServerName;database=DBName;user id=MyUserName;password=MyPassword;realm=comani.com;krb5conffile=/etc/krb5.conf;
+
+### Kerberos Parameters
+
+* `krb5conffile` - path to kerberos configuration file.
+* `realm` - Domain name for kerberos authentication.
```

Comment:
>realm` - Domain name for kerberos authentication. [](http://example.com/codeflow?start=3&length=49)

The JDBC driver docs have "realm" as optional

[https://docs.microsoft.com/sql/connect/jdbc/setting-the-connection-properties?view=sql-server-ver16](https://docs.microsoft.com/sql/connect/jdbc/setting-the-connection-properties?view=sql-server-ver16)


(Version 9.4+) The realm for Kerberos authentication. Set this value to override the Kerberos authentication realm the driver autodetects from the server's realm.

Do you know what it means by "autodetects from the server's realm"? Is that a feature in JDBC we are missing here? Or is it something handled in the krb module itself?

Reply:
will check up on the JDBC part and revert. I hope, I have answered rest of your questions?